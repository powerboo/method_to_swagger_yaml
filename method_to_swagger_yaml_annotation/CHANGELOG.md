## 0.0.3

- add IgnoreFieldInYaml annotation

## 0.0.2

- add useToString option in RequestParameter

## 0.0.1

- init
