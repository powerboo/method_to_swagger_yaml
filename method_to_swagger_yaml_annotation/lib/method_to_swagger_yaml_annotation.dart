library method_to_swagger_yaml_annotation;

class ConvertTargetClass {
  final String title;
  final String? description;
  final String version;
  final List<Tag> tags;

  const ConvertTargetClass({
    required this.title,
    required this.version,
    this.description,
    this.tags = const [],
  });
}

class ConvertTargetMethod {
  final HttpMethodDiv httpMethod;
  final String pathName;
  final List<String> tags;
  final String? summary;
  final String? operationId;

  /// Description for path
  final String? description;

  /// Description for HttpStatusCode 200
  final String responseDescription;

  const ConvertTargetMethod({
    required this.httpMethod,
    required this.pathName,
    this.description,
    this.responseDescription = "Successful operation",
    this.operationId,
    this.tags = const [],
    this.summary,
  });
}

class RequestParameter {
  final RequestParameterDiv requestParameterDiv;
  final String? description;
  final PathParameterStyleDiv? pathStyle;
  final QueryParameterStyleDiv? queryStyle;
  final bool? explode;
  final bool useToString;

  const RequestParameter({
    this.requestParameterDiv = RequestParameterDiv.query,
    this.description,
    this.pathStyle,
    this.queryStyle,
    this.explode,
    this.useToString = false,
  });
}

enum QueryParameterStyleDiv {
  from("from"),
  spaceDelimited("spaceDelimited"),
  pipeDelimited("pipeDelimited"),
  deepObject("deepObject"),
  ;

  final String toStringValue;

  const QueryParameterStyleDiv(this.toStringValue);
  factory QueryParameterStyleDiv.fromString({
    required String value,
  }) {
    final indexEnum = QueryParameterStyleDiv.values
        .firstWhere((e) => e.toStringValue == value);
    return indexEnum;
  }
}

enum PathParameterStyleDiv {
  simple("simple"),
  label("label"),
  matrix("matrix"),
  ;

  final String toStringValue;

  const PathParameterStyleDiv(this.toStringValue);
  factory PathParameterStyleDiv.from({
    required String value,
  }) {
    final indexEnum = PathParameterStyleDiv.values
        .firstWhere((e) => e.toStringValue == value);
    return indexEnum;
  }
}

enum RequestParameterDiv {
  query("query"),
  path("path"),
  /*
  TODO: implements
  header("header"),
  cookie("cookie"),
  // */
  ;

  final String toStringValue;

  const RequestParameterDiv(this.toStringValue);
  factory RequestParameterDiv.from({
    required String value,
  }) {
    final indexEnum =
        RequestParameterDiv.values.firstWhere((e) => e.toStringValue == value);
    return indexEnum;
  }
}

enum HttpMethodDiv {
  get("get"),
  post("post"),
  put("put"),
  patch("patch"),
  delete("delete"),
  ;

  final String toStringValue;

  const HttpMethodDiv(this.toStringValue);
  factory HttpMethodDiv.from({
    required String value,
  }) {
    final indexEnum =
        HttpMethodDiv.values.firstWhere((e) => e.toStringValue == value);
    return indexEnum;
  }
}

class ExternalDocs {
  final String? description;
  final String url;

  const ExternalDocs({
    required this.url,
    this.description,
  });
}

class ExternalDocsException implements Exception {
  late final String message;
  ExternalDocsException(final String message) {
    this.message = "[ExternalDocsException] $message";
  }
  @override
  String toString() {
    return message;
  }
}

class Tag {
  final String name;
  final String? description;
  final ExternalDocs? externalDocs;

  const Tag({
    required this.name,
    this.description,
    this.externalDocs,
  });
}

class TagException implements Exception {
  late final String message;
  TagException(final String message) {
    this.message = "[TagException] $message";
  }
  @override
  String toString() {
    return message;
  }
}
