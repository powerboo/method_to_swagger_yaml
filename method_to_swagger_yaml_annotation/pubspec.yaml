name: method_to_swagger_yaml_annotation
description: "method to swagger yaml annotation."
version: 0.0.1
repository: https://github.com/powerboo/method_to_swagger_yaml

environment:
  sdk: ">=3.2.4 <4.0.0"
