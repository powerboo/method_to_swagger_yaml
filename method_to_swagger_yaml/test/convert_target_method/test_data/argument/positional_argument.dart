import 'package:method_to_swagger_yaml_annotation/method_to_swagger_yaml_annotation.dart';

@ConvertTargetClass(
  title: "positional argument.",
  version: "0.0.1",
)
abstract interface class PositionalArgumentClass {
  @ConvertTargetMethod(
    httpMethod: HttpMethodDiv.get,
    pathName: "positional-argument",
  )
  Future<void> getMethod(
    String positionalStringValue,
    String? positionalNullableStringValue,
    PositionalArgument positionalClassValue,
    PositionalArgument? positionalNullableClassValue,
  );
}

class PositionalArgument {
  final String value;
  PositionalArgument({required this.value});
}
