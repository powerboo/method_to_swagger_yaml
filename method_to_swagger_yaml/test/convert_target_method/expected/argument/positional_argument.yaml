openapi: 3.0.0
info:
  title: positional argument.
  version: 0.0.1
paths:
  /positional-argument:
    get:
      operationId: getMethod
      parameters:
        - name: positionalStringValue
          in: query
          required: true
          schema:
            type: string
        - name: positionalNullableStringValue
          in: query
          required: false
          schema:
            type: string
        - name: positionalClassValue
          in: query
          required: true
          schema:
            $ref: '#/components/schemas/GetMethodPositionalClassValue'
        - name: positionalNullableClassValue
          in: query
          required: false
          schema:
            $ref: '#/components/schemas/GetMethodPositionalNullableClassValue'
      responses:
        '200':
          description: Successful operation
components:
  schemas:
    GetMethodPositionalStringValue:
      type: string
    GetMethodPositionalNullableStringValue:
      type: string
    GetMethodPositionalClassValue:
      type: object
      properties:
        positionalClassValue:
          type: object
          properties:
            value:
              type: string
    GetMethodPositionalNullableClassValue:
      type: object
      properties:
        positionalNullableClassValue:
          type: object
          properties:
            value:
              type: string
